src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs

```diff
@@ -1045,7 +1045,9 @@ private void ImportFieldAccess(int token, bool isStatic, string reason)
                     // magic fields the compiler synthetized, the data blob might bring more dependencies
                     // and we need to scan those.
                     _dependencies.Add(_compilation.GetFieldRvaData(field), reason);
-                    // TODO: lazy cctor dependency
+                    Debug.Assert(!field.OwningType.HasInstantiation);
```

Comment:
Are fields with RVAs disallowed in ECMA or rejected by CoreCLR? Or is this an unimplemented case?

Reply:
ECMA doesn't seem to disallow it but this code will crash if it's on a generic type; we need double the lines to handle that.

These would map to the same thing irrespective of what's the T and that feels weird enough that I didn't feel like writing handling/tests for it.